core/src/main/java/org/infinispan/configuration/parsing/ParserRegistry.java

```diff
@@ -321,7 +335,7 @@ public String toString() {
    public void serialize(ConfigurationWriter writer, String name, Configuration configuration) {
       writer.writeStartDocument();
       CoreConfigurationSerializer serializer = new CoreConfigurationSerializer();
-      serializer.writeCache(writer, name, configuration, true);
+      serializer.writeCache(writer, name, configuration, name == null || name.isBlank());
```

Comment:
Maybe we can remove the `unnamed` parameter from `writeCache` and check in there if the name is null/blank.